### PR TITLE
[Azure Cognitive Search][Resource Manager]: Add Swagger API for 2021-04-01-Preview API version

### PR DESCRIPTION
**Pull request by arv100kri**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

#### Description of changes

+ Adding new properties that can be included during search service creation/update
    + Encryption with CMK
    + Support for user assigned identity
    + Disabling data exfiltration scenarios
    + Support for new authentication options
    + Support for setting up semantic search sku

+ Added a new property to the response of the `Operations` API that indicates whether an operation is a data plane operation or not.

Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.  **End of May**
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. ** Mid-June**
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by mentat9** [comment]
When you are ready for ARM review, please upgrade your PR from Draft to Open.

**Review by tjacobhi — APPROVED** [review]

File: specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json [hunk 1]
```diff
@@ -0,0 +1,2441 @@
...
+    "NetworkRuleSet": {
+      "properties": {
+        "ipRules": {
+          "type": "array",
+          "description": "A list of IP restriction rules that defines the inbound network(s) with allowing access to the search service endpoint. At the meantime, all other public IP networks are blocked by the firewall. These restriction rules are applied only when the 'publicNetworkAccess' of the search service is 'enabled'; otherwise, traffic over public interface is not allowed even with any public IP rules, and private endpoint connections would be the exclusive access method.",
+          "items": {
+            "$ref": "#/definitions/IpRule"
+          }
+        },
+        "byPass": {
```
Comment (majastrz) [inline specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json:1520]: >byPass [](http://example.com/codeflow?start=9&length=6)

Is it "by pass" or "bypass". If it's the latter, you should fix the casing.

File: specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json [hunk 2]
```diff
@@ -0,0 +1,2441 @@
...
+          "type": "string",
+          "enum": [
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "SearchByPass",
```
Comment (majastrz) [inline specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json:1526]: >SearchByPass [](http://example.com/codeflow?start=21&length=12)

Same bypass casing question

File: specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json [hunk 3]
```diff
@@ -0,0 +1,2441 @@
...
+        }
+      },
+      "description": "Defines the SKU of an Azure Cognitive Search Service, which determines price tier and capacity limits.",
+      "externalDocs": {
+        "url": "https://azure.microsoft.com/documentation/articles/search-sku-tier/"
+      }
+    },
+    "DataPlaneAuthOptions": {
+      "properties": {
+        "apiKeyOnly": {
```
Comment (majastrz) [inline specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json:1377]: >apiKeyOnly [](http://example.com/codeflow?start=9&length=10)

Based on the description, these properties look like secrets. If so, please mark them with `x-ms-secret: true`.

File: specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json (hunk 1 above)
Comment (arv100kri) [reply specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json:1520]: It is the latter, thanks for catching it, will fix here and below!

File: specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json (hunk 3 above)
Comment (arv100kri) [reply specification/search/resource-manager/Microsoft.Search/preview/2021-04-01-preview/search.json:1377]: These are not secrets... this is an object which when present (i.e., non-null) indicates that the API key needs to be used for authentication... the actual "API key" itself is a secret, but this property just indicates whether that secret needs to be used...

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**